File: Nordeus-Challenge/Assets/_Scripts/Models/Items/ItemDefinition.cs
using System;

[Serializable]
public class ItemBonusStats
{
    public int health;
    public int attack;
    public int defense;
    public int magic;
}

[Serializable]
public class ItemDefinition
{
    public string id;
    public string name;
    public string description;
    public string itemType;
    public ItemBonusStats bonusStats;
    public PassiveEffect[] passiveEffects;

    public ItemType ItemType =>
        Enum.TryParse(itemType, out ItemType t) ? t : ItemType.ONE_HANDED;
}

File: backend/src/main/java/com/example/backend/models/ItemDefinition.java
package com.example.backend.models;

import lombok.Data;

import java.util.List;

@Data
public class ItemDefinition {
    private String id;
    private String name;
    private String description;
    private ItemType itemType;
    private CharacterStats bonusStats;
    private List<PassiveEffect> passiveEffects;
}
